### PR TITLE
Add `packageAndLockFilesOutOfSync` TurboSnap bail reason

### PR DESCRIPTION
**Pull request by codykaup**
TurboSnap can find two different dependency versions in the project's lock file (one for prod and a different one for dev) which breaks the lock file parsing. We currently silence those errors in the CLI and continue with the build process. However, we want to start alerting users to this discovery and suggest a fix for that in the CLI output.

<details>
  <summary>📦 Published PR as canary version: <code>11.16.2--canary.1111.11577136998.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install chromatic@11.16.2--canary.1111.11577136998.0
  # or 
  yarn add chromatic@11.16.2--canary.1111.11577136998.0
  ```
</details>

### CODE REVIEW

File: node-src/tasks/upload.test.ts [hunk 1]
```diff
@@ -273,6 +273,30 @@ describe('traceChangedFiles', () => {
...
+      http,
+      options: {},
+      sourceDir: '/static/',
+      fileInfo: { statsPath: '/static/preview-stats.json' },
+      git: { changedFiles: ['./example.js', './package.json'], packageMetadataChanges },
+      turboSnap: {},
+    } as any;
+    await traceChangedFiles(ctx, {} as any);
+
+    expect(ctx.turboSnap.bailReason).toEqual({ packageAndLockOutOfSync: true });
```
Comment (jwir3) [inline node-src/tasks/upload.test.ts:295]:
```suggestion
    expect(ctx.turboSnap.bailReason).toEqual({ packageAndLockFilesOutOfSync: true });
```

**Review by jwir3 — APPROVED** [review]
This looks good to me with one minor change that will allow tests to pass.

**Comment by jwir3** [comment]
Whoops... @codykaup I didn't see this was a draft PR. If you're still working on it and want review, just go ahead and ping me.

**Comment by codykaup** [comment]
> Whoops... @codykaup I didn't see this was a draft PR. If you're still working on it and want review, just go ahead and ping me.

There's some discussion on the other side (https://github.com/chromaui/chromatic/pull/9059) because the NPM lock file isn't friendly to this notification and could cause more harm than good.

**Comment by codykaup** [comment]
We're choosing to not do this right now. We may come back to this in the future.

**Pull request closed without merging**